apps/common/test/lexical/math_test.exs

```diff
@@ -28,4 +28,22 @@ defmodule Lexical.MathTest do
       assert Math.clamp(high, low, mid) == mid
     end
   end
+
+  describe "count_leading_space/1" do
+    test "returns 0 when string is empty" do
+      assert Math.count_leading_spaces("") == 0
+    end
+
+    test "returns 0 when string has no leading spaces" do
+      assert Math.count_leading_spaces("hello") == 0
+    end
+
+    test "returns 2 when string has two leading spaces" do
+      assert Math.count_leading_spaces("  hello") == 2
+    end
+
+    test "return 1 when string has one leading tab" do
+      assert Math.count_leading_spaces("\thello") == 1
+    end
```

Comment:
i feel like property tests would be appropriate here.

Reply:
Added one via e767c11b7c6fb7798e8c9788830b3c741aef4b38